tests/installation/yast2_firstboot.pm

```diff
@@ -87,6 +92,21 @@ sub firstboot_registration {
     $firstboot->press_next();
 }
 
+sub firstboot_keyboard {
+    my $firstboot = $testapi::distri->get_firstboot();
+    $firstboot->keyboard_page();
```

Comment:
There is a difference in the way we treat modules/clients in firstboot, as each client is adressed in a single module. So if I would like to mimic what is done in partitioner I would have to call the methods in my controller something like "check_keyboard_page_and_press_next" because otherwise the "press_next" method only presses next but does not do is_shown, since every page needs a differenet is_shown method. This is why I wanted to have a unique way of asserting pages at the first place, so that there would be only one "is_shown" checking a variable string passed to the method, using genericPage for all smoke tests. WDYT ?

Reply:
Otherwise I can have those methods to just call is_shown and call them just "check_lan_page" or "assert_lan_page"...

Reply:
ok, I see the difference, that we only have one controller for all the pages, when normally we have one controller per page.
So then the idea would be to name it, `setup_keyboard()` where we do nothing except press next (well ok, check that the page is there, but this is more our safety than functionality).
We also should use inheritance between pages for having available next button in keyboard page.
You can still reuse `$firstboot->press_next()` as a method of the controller:

In the controller:
```
sub setup_keyboard {
    my ($self) = @_;
    die 'blabla' unless $self->get_keyboard_page->is_shown();
    $self->get_keyboard_page->press_next();
}
```

Reply:
Yes, I like a naming like that with inheriting press_next.